### PR TITLE
io/lab: Update now we pipe

### PR DESCRIPTION
**Pull request by Alex-deVis**
Summary:

- Add sample output
- Replace `scanf` with `fgets`
- Add `Makefile` and `.gitignore` to solution folder

### CODE REVIEW

File: content/chapters/io/lab/solution/pipes/Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+BINARY = anonymous_pipe
```
Comment (teodutu) [inline content/chapters/io/lab/solution/pipes/Makefile:1]: Change this to a link to `../../support/pipes/Makefile`.

File: content/chapters/io/lab/solution/pipes/anonymous_pipe.c [hunk 2]
```diff
@@ -9,14 +9,15 @@
...
 #include "utils/utils.h"
 
-#define BUFSIZE 	128
-#define EXIT_STR	"exit"
-#define PIPE_READ	0
-#define PIPE_WRITE	1
+#define BUFSIZE 128
+#define EXIT_STR "exit"
+#define PIPE_READ 0
+#define PIPE_WRITE 1
```
Comment (teodutu) [inline content/chapters/io/lab/solution/pipes/anonymous_pipe.c:15]: The tabs were OK. It looks better to keep the `#define`s aligned.

File: content/chapters/io/lab/solution/pipes/anonymous_pipe.c [hunk 3]
```diff
@@ -50,11 +53,18 @@ static void parent_loop(int writefd)
...
+	while (1)
+	{
 		memset(input, 0, BUFSIZE);
-		scanf("%s",  input);
+		fgets(input, BUFSIZE, stdin);
+		// Remove trailing newline
+		if (input[strlen(input) - 1] == '\n')
+		{
+			input[strlen(input) - 1] = '\0';
+		}
```
Comment (teodutu) [inline content/chapters/io/lab/solution/pipes/anonymous_pipe.c:64]:
```suggestion
			input[strlen(input) - 1] = '\0';
```

File: content/chapters/io/lab/support/pipes/anonymous_pipe.c [hunk 4]
```diff
@@ -9,14 +9,15 @@
 
 #include "utils/utils.h"
 
-#define BUFSIZE 	128
-#define EXIT_STR	"exit"
-#define PIPE_READ	0
-#define PIPE_WRITE	1
+#define BUFSIZE 128
```
Comment (teodutu) [inline content/chapters/io/lab/support/pipes/anonymous_pipe.c:12]: Ditto: all comments from above.

**Review by teodutu — CHANGES_REQUESTED** [review]
The sample output is fine. I made some suggestions to the code.

File: content/chapters/io/lab/solution/pipes/anonymous_pipe.c (hunk 2 above)
Comment (Alex-deVis) [reply content/chapters/io/lab/solution/pipes/anonymous_pipe.c:15]: We should provide a `.clang-format` file so we could correct this automatically.

File: content/chapters/io/lab/solution/pipes/anonymous_pipe.c [hunk 5]
```diff
@@ -9,116 +9,115 @@
...
+#define EXIT_STR   "exit"
+#define PIPE_READ  0
+#define PIPE_WRITE 1
+
+static bool check_for_exit(const char *input) {
+    if (strcmp(input, EXIT_STR) == 0 || strlen(input) == 0) {
+        return true;
+    }
+
+    return false;
```
Comment (teodutu) [inline content/chapters/io/lab/solution/pipes/anonymous_pipe.c:22]: Why change the coding style and why use spaces instead of tabs. Revert this.

**Review by teodutu — CHANGES_REQUESTED** [review]
I guess you used some auto-formatting tool that messed up the code.

File: content/chapters/io/lab/solution/pipes/anonymous_pipe.c [hunk 6]
```diff
@@ -9,116 +9,115 @@
...
-{
-	if (strcmp(input, EXIT_STR) == 0) {
-		return true;
-	}
-
-	return false;
+#define BUFSIZE    128
+#define EXIT_STR   "exit"
+#define PIPE_READ  0
+#define PIPE_WRITE 1
```
Comment (teodutu) [inline content/chapters/io/lab/solution/pipes/anonymous_pipe.c:15]: Tabs not spaces, the way it was before your changes.

File: content/chapters/io/lab/solution/pipes/anonymous_pipe.c (hunk 2 above)
Comment (teodutu) [reply content/chapters/io/lab/solution/pipes/anonymous_pipe.c]: `checkpatch.pl` can do this IIRC.

**Review by teodutu — APPROVED** [review]
:+1:

**Pull request merged**